librz/core/canalysis.c

```diff
@@ -5244,18 +5244,19 @@ static void *analysis_op_next(RzIterator *it) {
 		return NULL;
 	}
 
-	ut64 addr = ctx->core->offset + ctx->offset_block;
-	if (ctx->max_op_size + addr > ctx->core->offset + ctx->core->blocksize) {
-		if (!rz_core_seek(ctx->core, addr, true)) {
+	if (ctx->max_op_size + ctx->offset_block >= ctx->core->blocksize && ctx->len > ctx->core->blocksize) {
+		if (!rz_core_seek(ctx->core, ctx->core->offset + ctx->offset_block, true)) {
 			return NULL;
 		}
 		ctx->offset_block = 0;
 	}
+	ut64 addr = ctx->core->offset + ctx->offset_block;
 	ut8 *ptr = ctx->core->block + ctx->offset_block;
```

Comment:
Since you are changing it anyway, I wonder if maybe it's better to convert it to transparent access? See https://github.com/rizinorg/rizin/issues/2695 for details. You can see https://github.com/rizinorg/rizin/pull/4023 as an example.

Reply:
Since `rz_core_io_read_at` is not implemented, I just removed the `core->block` dependency.

Reply:
Yes, that's what I meant.